### DISCUSSION TITLE
MTU problems with K3S deployed over wireguard

### DISCUSSION THREAD

**Discussion by WeaselScience** [k3s-io/k3s#9315] [Q&A]
I've the following situation: three physical locations, one K3S node in each of them. Each location is a secure LAN, and the gateways connect all three LANs together using Wireguard.

I'm having issues similar to https://github.com/k3s-io/k3s/issues/9097

My question is: how do i properly and permanently reduce the MTU of the vxlan interfaces?

**Comment by brandond** [comment]
Have you considered using the native wireguard backend, instead of using vxlan over wireguard?

I'm not sure there is currently an option to override the flannel interface MTU. It copies the MTU from the outbound interface; if for some reason it's not detecting the interface correctly you can try setting the --flannel-iface option.

**Comment by WeaselScience** [comment]
I'd prefer not to use the wireguard backend, as then i'd have two wireguard instances running - one for lan to lan communication, and one for node to node. It's possible, but i'd rather not do port forwarding just for k3s.

The wireguard interface is not accessible to the actual k3s nodes - it's handled transparently to the nodes by my gateways, so flannel has no way of knowing that the acceptable MTU is less than 1450.

The method i decided to go for was changing the MTU on the LAN interfaces of my k3s nodes to 1420. This forced flannel to set it's MTU to 1370, which made everything work swimmingly.
